packages/engine/Source/Scene/VoxelContent.js

```diff
@@ -1,167 +1,202 @@
-    const end = start + bufferViewJson.byteLength;
-    const buffer = buffersU8[bufferViewJson.buffer];
-    const bufferView = buffer.subarray(start, end);
-    bufferViewsU8[i] = bufferView;
+  return new VoxelContent({ loader });
+};
+
+/**
+ * Updates the content until all resources are ready for rendering.
+ * @param {FrameState} frameState The frame state
+ * @private
+ */
+VoxelContent.prototype.update = function (primitive, frameState) {
+  const loader = this._loader;
+
+  if (this._ready) {
+    // Nothing to do
+    return;
   }
 
-  const propertyTableIndex = chunks.json.voxelTable;
-  const propertyTableJson = chunks.json.propertyTables[propertyTableIndex];
+  // Ensures frames continue to render in requestRender mode while resources are processing
+  frameState.afterRender.push(() => true);
 
-  const content = new VoxelContent(resource);
+  if (!defined(loader)) {
+    this._ready = true;
+    return;
+  }
 
-  content._metadataTable = new MetadataTable({
-    count: propertyTableJson.count,
-    properties: propertyTableJson.properties,
-    class: metadataSchema.classes[propertyTableJson.class],
-    bufferViews: bufferViewsU8,
-  });
+  if (this._resourcesLoaded) {
+    const { attributes } = loader.components.scene.nodes[0].primitives[0];
+    this._metadata = processAttributes(attributes, primitive);
+    this._ready = true;
+    return;
+  }
 
-  return content;
+  this._resourcesLoaded = loader.process(frameState);
 };
 
-function requestBuffers(resource, json, binary) {
-  const buffersLength = json.buffers.length;
-  const bufferPromises = new Array(buffersLength);
-  for (let i = 0; i < buffersLength; i++) {
-    const buffer = json.buffers[i];
-    if (defined(buffer.uri)) {
-      const baseResource = resource;
-      const bufferResource = baseResource.getDerivedResource({
-        url: buffer.uri,
-      });
-      bufferPromises[i] = bufferResource
-        .fetchArrayBuffer()
-        .then(function (arrayBuffer) {
-          return new Uint8Array(arrayBuffer);
-        });
-    } else {
-      bufferPromises[i] = Promise.resolve(binary);
+/**
+ * Processes the attributes from the glTF loader, reordering them into the order expected by the primitive.
+ *
+ * @param {ModelComponents.Attribute[]} attributes The attributes to process
+ * @param {VoxelPrimitive} primitive The primitive for which this voxel content will be used.
+ * @returns {TypedArray[]} An array of typed arrays containing the attribute values
+ * @private
+ */
+function processAttributes(attributes, primitive) {
+  const { names, types, componentTypes } = primitive.provider;
+  const data = new Array(attributes.length);
+
+  for (let i = 0; i < attributes.length; i++) {
+    // The attributes array from GltfLoader is not in the same order as
+    // names, types, etc. from the provider.
+    // Find the appropriate glTF attribute based on its name.
+    // Note: glTF custom attribute names are prefixed with "_"
+    const name = `_${names[i]}`;
```

Comment:
Could this change to retrieve the property names from the glTF's property attribute? The attribute for a `temperature` property might be named `_TEMPERATURE`, which will mess with the `find` function below.